docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,215 @@
+configure a max retention period. Some automatic purging mechanism is triggered
+when logs are older than the retention period or when the persistent storage is
+about to reach its capacity limit. Purging jobs are logged.
+A typical and default retention period is of 2 weeks.
+A formula can be used by solution developers in order to properly size the
+persistent storage for log centralization (cf documentation requirement).
+
+*Horizontally scalable (capacity and performance)*
+
+The Platform Administrator can scale the service in order to ingest and store
+more logs. It can be because more workloads are running on the platform or
+because there is a need to keep bigger history of logs.
+
+*Highly Available*
+
+Log collection, storage and browsing services can be replicated in order to
+ensure the same level of High Availability Prometheus has i.e. can lose all
+replicas but one.
+
+*Log Querying*
+
+Get all logs for a given period, node(s), pod regex, limited list of predefined
+labels and free keywords text.
+Typical Zenko use case: collecting all logs across several components, related
+to a S3 uniq request.
+Typical predefined labels are severity and namespace.
+
+*Log statistics (nice to have)*
+
+The Log centralization service also offers the ability to consume statistics
+about the logs like the number of occurrences of one type of log during a
+certain period of time.
+
+*Monitorable/Observable service (health, performances and alerts)*
+
+The Platform Administrator can monitor capacity usage, ingestion rate, IOPS,
+latency and bandwidth of the Log centralization service. He can also monitor
+the health of the service (i.e. if some active alerts exist).
+He is notified through an alert notification when the service is degraded or
+unavailable. It can be because the persistent storage is full or unhealthy or
+because the service does not manage to ingest logs at the requested pace.
+
+Here are few example of situations we would like to detect through those KPIs:
+
+- a workload generating a crazy amount of logs
+- a burst of ingested logs
+- the log persistent storage getting full
+- very slow api responses (impacting usability in Grafana dashboards)
+- the ingestion of logs working too slowly
+
+*Performances (TBD)*
+
+Typical workloads can generate around 1000 logs per second per node.
+
+User Stories
+------------
+
+- As a Platform Administrator, I want to browse all MetalK8s containers logs
+  (from all servers) from a unique endpoint, in order to ease distributed
+  K8s and MetalK8s services error investigation.
+
+- As a Platform Administrator, I want to browse non container (kubelet,
+  containerd, salt-minion, cron) logs (from all servers) from a
+  unique endpoint, in order to ease System error investigation.
+
+- As a Storage Administrator, I want to browse all Solution instance containers
+  logs (from all servers) from a unique endpoint, in order to ease Solution
+  instance error investigation.
```

Comment:
NOTE that if you are "horizontally scalable" you will likely not be able to have all servers logs from only one endpoint as we do not have one server storing all servers data it's spread in this case.
So logs will come from different data sources
(At least it's the behavior for prometheus by default and it will likely be the same with Loki&co)